File: autonova/src/main/java/com/example/demo/service/VehiculoService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.modelo.Vehiculos;

public interface VehiculoService {
	
	public List<Vehiculos> getDisponiblesPorTipoYestado(String tipo, String estado);
	
	Vehiculos getplaca(String placa);
	
	Vehiculos actualizarEstado(long idVehiculo,String estado);

}

File: AutoNova_F/src/app/modules/Vehiculos/services/vehiculo.service.ts
import { Injectable } from '@angular/core';
import { Vehiculo } from '../entidades/vehiculo';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

const BASE_URL="http://localhost:8085/vehiculos";
@Injectable({
  providedIn: 'root'
})
export class ServiceService {

  constructor(private http:HttpClient) { 

  
  }
   getVehiculosDisponibles(tipo: string, estado: string): Observable<Vehiculo[]> {
    const url = `${BASE_URL}/verdisponibles?tipo=${tipo}&estado=${estado}`;
    return this.http.get<Vehiculo[]>(url);
  }

buscarporplaca(placa: string): Observable<Vehiculo> {  
  return this.http.get<Vehiculo>(`${BASE_URL}/buscarporplaca?placa=${placa}`);
}
 cambiarEstadoAVehiculoDisponible(idVehiculo: number): Observable<any> {
  return this.http.put(
    `${BASE_URL}/actualizarEstado?idVehiculo=${idVehiculo}&estado=disponible`,
    {}
  );
}
}
